### ISSUE TITLE
[FEATURE] : UML for Typescript

### ISSUE THREAD

**Issue by leocaseiro** [compodoc/compodoc#313]
That would be nice be able to see the diagram of some classes and interfaces 

An example is within https://tsuml-demo.firebaseapp.com/

![screen shot 2017-09-21 at 1 18 05 pm](https://user-images.githubusercontent.com/940070/30677503-5ae4bfb8-9ecf-11e7-9896-32cede1f7ba6.png)

I believe this library should help https://github.com/joaompneves/tsviz

**Comment by vogloblinsky** [comment]
@leocaseiro 
Yes, it is an interesting idea. tsviz seems cool, but the dependency of graphviz is a quite complicated for having a very simple CLI tool to use.

**Comment by krisztianb** [comment]
Yes, it would be really helpful to at least have a class diagramm (showing maybe only the base class and the derived classes) on the page of a class. I think doxygen/graphviz has this including an image map, so that you can jump to the related classes by clicking on them in the diagramm.

**Comment by gchaviaras** [comment]
@leocaseiro 
Apparently the webpage that you embed is an implementation demo of this lib (https://github.com/fsahmad/typescript-uml)

@vogloblinsky There is a way to generate png/svg images from plantuml files using this public server http://www.plantuml.com/plantuml/png/. With this way you remove the dependencies of graphviz.

A complete example ( not very good though ) of using **typescript-uml** combined with the public plantuml server is this tool https://www.npmjs.com/package/tsuml-cli (bin/tsuml-cli.js:106)

**Comment by krisztianb** [comment]
Bzzzz bot bzzzzz... This feature is still very relevant. It's just that it will be implemented in a later milestone.
